siphon/simplewebservice/wyoming.py

```diff
@@ -0,0 +1,149 @@
+from metpy.calc import get_wind_components
+
+
+def get_upper_air_data(time, site_id):
+    r"""Download and parse upper air observations from an online archive.
+
+    Parameters
+    ----------
+    time : datetime
+        The date and time of the desired observation.
+
+    site_id : str
+        The three letter ICAO identifier of the station for which data should be
+        downloaded.
+
+    kwargs
+        Arbitrary keyword arguments to use to initialize source
+
+    Returns
+    -------
+        :class:`pandas.DataFrame` containing the data
+
+    """
+    endpoint = WyomingUpperAir('http://weather.uwyo.edu/')
+    fobj = endpoint.get_data(time, site_id)
+    info = endpoint.parse(fobj)
+
+    direction, spd, spd_units = info['wind']
+    u, v = get_wind_components(spd, np.deg2rad(direction))
+
+    def insert_with_units(data_dict, name, data, units):
+        data_dict[name] = data
+        data_dict[name + '_units'] = units
+        return data_dict
+
+    data = dict()
+    for key, name in [('p', 'pressure'),
+                      ('z', 'height'),
+                      ('t', 'temperature'),
+                      ('td', 'dewpoint')]:
+        key_data, key_units = info[key]
+
+        insert_with_units(data, name, key_data, key_units)
+
+    return pd.DataFrame(data=data)
+
+
+class WyomingUpperAir(HTTPEndPoint):
+    """Download and parse data from the University of Wyoming's upper air archive."""
+
+    def get_data(self, time, site_id, region='naconf'):
+        """Download data from the University of Wyoming's upper air archive.
+
+        Parameters
+        ----------
+        time : datetime
+            Date and time for which data should be downloaded
+        site_id : str
+            Site id for which data should be downloaded
+        region : str
+            The region in which the station resides. Defaults to `naconf`.
+
+        Returns
+        -------
+        a file-like object from which to read the data
+
+        """
+        path = ('cgi-bin/sounding?region={region}&TYPE=TEXT%3ALIST'
+                '&YEAR={time:%Y}&MONTH={time:%m}&FROM={time:%d%H}&TO={time:%d%H}'
+                '&STNM={stid}').format(region=region, time=time, stid=site_id)
+
+        resp = self.get_path(path)
+
+        # See if the return is valid, but has no data
+        if resp.text.find('Can\'t') != -1:
+            raise ValueError(
+                'No data available for {time:%Y-%m-%d %HZ} from region {region} '
+                'for station {stid}.'.format(time=time, region=region,
+                                             stid=site_id))
+
+        # Since the archive text format is embedded in HTML, look for the <PRE> tags
+        data_start = resp.text.find('<PRE>')
+        data_end = resp.text.find('</PRE>', data_start)
+
+        # Grab the stuff *between* the <PRE> tags -- 6 below is len('<PRE>\n')
+        buf = resp.text[data_start + 6:data_end]
+        return BytesIO(buf.encode(encoding='UTF-8'))
```

Comment:
Can probably just move the contents of `parse` here since we're not trying to have the same common structure as we did in MetPy. Might be able to leverage pandas' own parsing code to do this work too.